src/catalog/database_catalog.cpp

```diff
@@ -1251,7 +1284,8 @@ index_oid_t DatabaseCatalog::GetIndexOid(const common::ManagedPointer<transactio
 
 const IndexSchema &DatabaseCatalog::GetIndexSchema(const common::ManagedPointer<transaction::TransactionContext> txn,
                                                    index_oid_t index) {
-  auto ptr_pair = GetClassSchemaPtrKind(txn, static_cast<uint32_t>(index));
+  transaction::TransactionContext fake_txn(transaction::timestamp_t{UINT64_MAX}, transaction::timestamp_t{UINT64_MAX});
```

Comment:
This will also be communicated in the design doc. Essentially, we want to use a fake txn in order to get any indices that are logically invisible to us (e.g. whose commit timestamp is after our start timestamp) so that we can still insert into them and keep them up-to-date.

Is this safe? Yes, from an MVCC viewpoint indices are performance hints anyways and are not going to affect the result. From a logical viewpoint we're only ever reading from live indices which are guaranteed to be correct (even if the commit timestamp is later than our sstart timestamp)